payjoin/src/core/send/v2/mod.rs

```diff
@@ -378,15 +380,29 @@ pub(crate) fn serialize_v2_body(
     output_substitution: OutputSubstitution,
     fee_contribution: Option<AdditionalFeeContribution>,
     min_fee_rate: FeeRate,
-) -> Result<Vec<u8>, CreateRequestError> {
+) -> Result<[u8; PADDED_PLAINTEXT_A_LENGTH], CreateRequestError> {
     // Grug say localhost base be discarded anyway. no big brain needed.
     let base_url = Url::parse("http://localhost").expect("invalid URL");
 
     let placeholder_url =
         serialize_url(base_url, output_substitution, fee_contribution, min_fee_rate, Version::Two);
     let query_params = placeholder_url.query().unwrap_or_default();
     let base64 = psbt.to_string();
-    Ok(format!("{base64}\n{query_params}").into_bytes())
+
+    let mut buf = [0u8; PADDED_PLAINTEXT_A_LENGTH];
+    let body_len = base64.len() + 1 + query_params.len();
```

Comment:
body length has `+ 1` to account for the `\n` newline byte in the plaintext length